### PR TITLE
feat: use team id as maintainers

### PR DESCRIPTION
**Pull request by IronCore864**
This PR changes the previous "owners", a list of GitHub user IDs, to "maintainers", a string of GitHub team ID. Assigning issues to a team is not supported, so an extra feature is added where user IDs are fetched from team ID.

More background:

Per discussions in [this issue](https://github.com/canonical/charm-relation-interfaces/pull/168) (see comment [here](https://github.com/canonical/charm-relation-interfaces/pull/168#issuecomment-2314981570) and [here](https://github.com/canonical/charm-relation-interfaces/pull/168#issuecomment-2316508980)) and an email thread, we have agreed to the followings:

- Use "maintainers" instead of "owners", because "owners" indicates ownership while the interface is meant to be "owned" by a group around it.
- Put a GitHub team ID as "maintainers", instead of a list of users. This eliminates the overhead of maintaining the list of user IDs. The purpose of this field is to assign issues to users when a certain interface test fails; although it's not supported to assign an issue to a team, this can be worked out in the [`charm-relation-interfaces`](https://github.com/canonical/charm-relation-interfaces) repo's [`run_matrix.py`](https://github.com/canonical/charm-relation-interfaces/blob/main/run_matrix.py) script.

This PR to `charm-relation-interfaces` reflects the above changes, see the other part in the [`pytest-interface-tester`](https://github.com/canonical/pytest-interface-tester) repo. Before merging into the main branch, it can’t be tested using GitHub Actions, but I’ve tested it in a forked repo in my own org, see [here](https://github.com/IronCoreWorks/issue-test/issues/5).

### CODE REVIEW

**Comment by IronCore864** [comment]
Now that the [PR for pytest-interface-tester](https://github.com/canonical/pytest-interface-tester/pull/22) is merged, this PR can be merged to use the latest `pytest-interface-tester`, v3.1.1, that has the "maintainer" feature.

File: run_matrix.py [hunk 1]
```diff
@@ -344,8 +344,22 @@ def test_failed(role_result: "_ResultsPerRole"):
...
+    team = gh.get_organization("canonical").get_team_by_slug(slug)
+    return (
+        [
+            member.login
+            for member in team.get_members()
+            if not member.login.endswith("-bot")
+        ]
+        if team
+        else []
+    )
```
Comment (benhoyt) [inline run_matrix.py:358]: This is a bit of a multi-line mouthful. How about this:

```python
if not team:
    return []
return [m.login for m in team.get_members() if not m.login.endswith("-bot")]
```

**Review by benhoyt — APPROVED** [review]
Looks good. One code style suggestion.

**Comment by IronCore864** [comment]
Merging this PR now, since pytest-interface-tester 3.2.0 has been released with the maintainer feature, and the tests have been fixed.

**Pull request merged**